libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2271,6 +2271,23 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 			}
 		}
 	}
+	else if (baseType.category() == Type::Category::InlineArray)
+	{
+		InlineArrayType const& arrayType = dynamic_cast<InlineArrayType const&>(baseType);
+		TupleExpression const& tuple = dynamic_cast<TupleExpression const&>(_indexAccess.baseExpression());
+
+		Expression const* index = _indexAccess.indexExpression();
+		Type const* indexType = index->annotation().type;
+		auto numberType = dynamic_cast<RationalNumberType const*>(indexType);
+		size_t literalIndex = numberType->literalValue(nullptr).convert_to<size_t>();
```

Comment:
Haha, ok... this is a problem :-).
```
function f(uint idx) {
  return [1,2,3][idx];
}
```
is perfectly valid (and works on ``develop``), i.e. you can't assume that we know the index :-)... and that's not even really too crazy of a use case - one might validly do something like that for a lookup table...

But keeping the inline array on stack does not work well together with dynamic index accesses...

Reply:
In general, we will probably need to convert to a statically sized memory array here and use a regular array index access on it... at least I don't see a better way out of my head...

Reply:
I really didn't expect this code to be final or even working for all scenarios. But, still, wanted to have any confirmation that I have any understanding of the code generation part.
Anyway, thank you for your suggestion on conversion to a memory array! I will try to do so :)

Reply:
Sure, it's a draft after all, but I saw the commit notification and had a quick peek nonetheless :-). And first thought this may be a reason for not keeping things on stack after all - but just converting to memory for the index access should be fine!